darwin/importer/importer.py

```diff
@@ -2312,3 +2330,189 @@ def _split_payloads(
         payloads.append(current_payload)
 
     return payloads
+
+
+def _get_remote_files_targeted_by_import(
+    importer: Callable[[Path], Union[List[dt.AnnotationFile], dt.AnnotationFile, None]],
+    file_paths: List[PathLike],
+    dataset: "RemoteDataset",
+    console: Optional[Console] = None,
+    use_multi_cpu: bool = True,
+    cpu_limit: int = 1,
+) -> List[DatasetItem]:
+    """
+    Parses local annotations files for import and returns a list of remote dataset items
+    targeted by the import. Handles chunking of requests if there are many files to
+    avoid URL length issues.
+
+    Parameters
+    ----------
+    importer: Callable[[Path], Union[List[dt.AnnotationFile], dt.AnnotationFile, None]]
+        The importer used to parse local annotation files
+    file_paths: List[PathLike]
+        A list of local annotation files to be uploaded
+    dataset: RemoteDataset
+        The remote dataset to fetch files from
+    console: Optional[Console]
+        The console object
+    use_multi_cpu: bool
+        Whether to use multi-CPU processing
+    cpu_limit: int
+        The number of CPUs to use for processing
+
+    Returns
+    -------
+    List[DatasetItem]
+        A list of remote dataset items targeted by the import
+
+    Raises
+    ------
+    ValueError
+        If no files could be parsed or if the URL becomes too long even with minimum chunk size
+    """
+    maybe_parsed_files = _find_and_parse(
+        importer, file_paths, console, use_multi_cpu, cpu_limit
+    )
+    if not maybe_parsed_files:
+        raise ValueError("Not able to parse any files.")
+
+    remote_filenames = list({file.filename for file in maybe_parsed_files})
+    remote_filepaths = [file.full_path for file in maybe_parsed_files]
+
+    chunk_size = 100
+    all_remote_files: List[DatasetItem] = []
+    while chunk_size > 0:
+        try:
+            for i in range(0, len(remote_filenames), chunk_size):
+                chunk = remote_filenames[i : i + chunk_size]
+                remote_files = dataset.fetch_remote_files(filters={"item_names": chunk})
+                all_remote_files.extend(remote_files)
+            break
+        except RequestEntitySizeExceeded:
+            chunk_size -= 8
+            if chunk_size <= 0:
+                raise ValueError(
+                    "Unable to fetch remote file list - URL too long even with minimum chunk size."
+                )
```

Comment:
Instead of this logic with errors and retries, you can just write function that ads file name to the query until its size is big enough but not over the limit. It would be clearer.